dolphin/alert_manager/trap_receiver.py

```diff
@@ -43,6 +45,106 @@ def __init__(self, trap_receiver_address, trap_receiver_port,
         self.trap_receiver_port = trap_receiver_port
         self.snmp_mib_path = snmp_mib_path
 
+    def sync_snmp_config(self, ctxt, snmp_config_to_del=None,
+                         snmp_config_to_add=None):
+        if snmp_config_to_del is not None:
+            self._delete_snmp_config(ctxt, snmp_config_to_del)
+
+        if snmp_config_to_add is not None:
+            self._add_snmp_config(ctxt, snmp_config_to_add)
+
+    def _add_snmp_config(self, ctxt, new_config):
+        LOG.info("Add snmp config:%s" % new_config)
+        storage_id = new_config["storage_id"]
+        version_int = self._get_snmp_version_int(ctxt, new_config["version"])
+        if version_int == 1 or version_int == 2:
+            community_string = new_config["community_string"]
+            community_index = self._get_community_index(storage_id)
+            config.addV1System(self.snmp_engine, community_index,
+                               community_string, contextName=community_string)
+        else:
+            username = new_config["username"]
+            engine_id = new_config["engine_id"]
+            auth_key = new_config.get("auth_key", None)
+            auth_protocol = new_config.get("auth_protocol", None)
+            privacy_key = new_config.get("privacy_key", None)
+            privacy_protocol = new_config.get("privacy_protocol", None)
+            if auth_key is not None:
+                auth_key = cryptor.decode(auth_key)
+            if privacy_key is not None:
+                privacy_key = cryptor.decode(privacy_key)
+            config.addV3User(
+                self.snmp_engine,
+                userName=username,
+                authKey=auth_key,
+                privKey=privacy_key,
+                authProtocol=self._get_usm_auth_protocol(ctxt,
+                                                         auth_protocol),
+                privProtocol=self._get_usm_priv_protocol(ctxt,
+                                                         privacy_protocol),
+                securityEngineId=v2c.OctetString(hexValue=engine_id))
+
+    def _delete_snmp_config(self, ctxt, snmp_config):
+        LOG.info("Delete snmp config:%s" % snmp_config)
+        version_int = self._get_snmp_version_int(ctxt, snmp_config["version"])
+        if version_int == 3:
```

Comment:
Can we define some macros for version in constants

Reply:
Done!